Tribler/community/multichain/community.py

```diff
@@ -293,34 +293,97 @@ def get_statistics(self, public_key=None):
+        :param amount: the amount of data transferred over this edge
+        """
+        if from_pk in edges:
+            edges[from_pk][to_pk] = edges[from_pk][to_pk] + amount if to_pk in edges[from_pk] else amount
+        else:
+            edges[from_pk] = {to_pk: amount}
+
     @blocking_call_on_reactor_thread
     def get_graph(self, public_key=None, neighbor_level=1):
         """
         Return a dictionary with the neighboring nodes and edges of a certain focus node within a certain radius,
-        regarding the local multichain database.
+        regarding the local MultiChain database.
 
         :param public_key: the public key of the focus node in raw format
         :param neighbor_level: the radius within which the neighbors have to be returned
-        :return: a tuple of a dictionary with nodes and a dictionary with edges
+        :return: a tuple of a list with nodes and a list with edges
         """
         if public_key is None:
             public_key = self.my_member.public_key
-        list_of_nodes = self.get_list_of_nodes(public_key, neighbor_level)
-        nodes = []
-        for current_key in list_of_nodes:
-            # TODO: retrieve more information at once when appropriate queries are present in database_driver
-            total_up, total_down = self.persistence.total_traffic(current_key)
-            nodes.append({"public_key": hexlify(current_key), "total_up": total_up, "total_down": total_down})
-            self.page_rank.add_node(hexlify(current_key))
-        edges = self.get_edges(list_of_nodes)
-        for edge in edges:
-            self.page_rank.add_edge(edge['from'], edge['to'], edge['amount'])
+
+        query_result = self.persistence.get_graph_edges(public_key, neighbor_level)
+        nodes = {hexlify(public_key): self.get_node(hexlify(public_key), {})}
+        edges = {}
+
+        for edge in query_result:
+            from_pk = hexlify(edge[0])
+            to_pk = hexlify(edge[1])
+            amount_up = edge[2]
+            amount_down = edge[3]
+            nodes[from_pk] = self.get_node(from_pk, nodes, total_up=edge[4], total_down=edge[5])
+            nodes[to_pk] = self.get_node(to_pk, nodes)
+            self.update_edges(from_pk, to_pk, edges, amount=amount_up)
+            self.update_edges(to_pk, from_pk, edges, amount=amount_down)
+
+        return_nodes = nodes.values()
+        return_edges = []
+        for node in nodes.keys():
+            self.page_rank.add_node(node)
+        for from_pk, edge_list in edges.iteritems():
+            new_edges = [{"from": from_pk, "to": to_pk, "amount": amount} for to_pk, amount in edge_list.iteritems()]
+            for edge in new_edges:
+                self.page_rank.add_edge(edge['from'], edge['to'], edge['amount'])
+            return_edges = return_edges + new_edges
+
         self.get_page_ranks()
-        for dic in nodes:
+        for dic in return_nodes:
             dic["page_rank"] = 0
```

Comment:
You could write this as `dic["page_rank"] = self.ranks[dic["public_key"]] if dic["public_key"] in self.ranks else 0`.